dev/io.openliberty.microprofile.openapi.2.0.internal_fat/fat/src/io/openliberty/microprofile/openapi20/fat/deployments/MergeServerXMLTestWithUpdate.java

```diff
@@ -0,0 +1,267 @@
+
+        List<String> failedToStop = new ArrayList<>();
+        for (String app : deployedApps) {
+            if (server.waitForStringInLogUsingMark("CWWKZ0009I:.*" + app) == null) {
+                failedToStop.add(app);
+            }
+        }
+
+        if (!failedToStop.isEmpty()) {
+            throw new AssertionError("The following apps failed to stop: " + failedToStop);
+        }
+
+    }
+
+    @Test
+    public void testTwoWarsWithUpdate() throws Exception {
+        System.out.println("testTwoWars - begin");
+        WebArchive war1 = ShrinkWrap.create(WebArchive.class, "test1.war")
+                                    .addClasses(DeploymentTestApp.class, DeploymentTestResource.class);
+
+        WebArchive war2 = ShrinkWrap.create(WebArchive.class, "test2.war")
+                                    .addClasses(DeploymentTestApp.class, DeploymentTestResource.class);
+
+        WebArchive war3 = ShrinkWrap.create(WebArchive.class, "test3.war")
+                                    .addClasses(DeploymentTestApp.class, DeploymentTestResource.class);
+
+        deployApp(war1);
+
+        //Swap the server.xml from saying everything is excluded to saying its included
+        logOnServer("/test1/log", "testTwoWarsUpdate1");
+        server.swapInServerXMLFromPublish(APP1_XML_NAME);
+        server.waitForStringInLogUsingMark("CWWKG0017I"); //The server configuration was successfully updated
+
+        assertRest("/test1/test");
+
+        // With one app deployed, we should only have the paths from war1 listed
+        String doc = OpenAPIConnection.openAPIDocsConnection(server, false).download();
+        JsonNode openapiNode = OpenAPITestUtil.readYamlTree(doc);
+        OpenAPITestUtil.checkPaths(openapiNode, 1, "/test");
+        assertEquals("/test1/test", OpenAPITestUtil.expandPath(openapiNode, "/test"));
+
+        // check that merge is not traced
+        assertThat(server.findStringsInLogsAndTraceUsingMark("Merged document:"), hasSize(0));
+        assertThat(server.findStringsInLogsAndTraceUsingMark("OpenAPIProvider retrieved from cache"), hasSize(0));
+
+        // check merged model was cached and cache is used on subsequent requests
+        OpenAPIConnection.openAPIDocsConnection(server, false).download();
+        assertThat(server.findStringsInLogsAndTraceUsingMark("Merged document:"), hasSize(0));
+        assertThat(server.findStringsInLogsAndTraceUsingMark("OpenAPIProvider retrieved from cache"), hasSize(1));
+
+        deployApp(war2);
+        deployApp(war3);
+        assertRest("/test1/test");
+        assertRest("/test2/test");
+        assertRest("/test3/test");
+        //Swap the server.xml from saying everything is excluded to saying its included
+        logOnServer("/test1/log", "testTwoWarsUpdate2");
+        server.swapInServerXMLFromPublish(SERVER_XML_NAME);
+        server.waitForStringInLogUsingMark("CWWKG0017I"); //The server configuration was successfully updated
```

Comment:
```suggestion
        server.setMarkToEndOfLog();
        server.swapInServerXMLFromPublish(SERVER_XML_NAME);
        server.waitForConfigUpdateInLogUsingMark(null);
```
I think we need to manually set the mark ourselves before starting the update if we want to use it afterwards.

Also there's a specific method for waiting for config updates.